pgmpy/prediction/NaiveIVRegressor.py

```diff
@@ -0,0 +1,212 @@
+        tags.regressor_tags.poor_score = True
+        return tags
+
+    def _prepare_feature_df(self, X, required_features=None) -> pd.DataFrame:
+        """
+        Accept a numpy/pandas dataframe and returns pandas df
+        If numpy array is passed, it converts to pandas df
+        """
+
+        if required_features is None:
+            required_features = self.feature_columns_fit_
+
+        # Step 2: Convert input to DataFrame format
+        if isinstance(X, pd.DataFrame):
+            X_df = X
+        else:
+            # For numpy arrays, use range index as column names
+            X_arr = np.asarray(X)
+            if X_arr.ndim == 1:
+                raise ValueError(
+                    "Reshape your data: X must be 2D. If using a 1D array, reshape it to (n_samples, 1)."
+                )
+            X_df = pd.DataFrame(X_arr, columns=range(X_arr.shape[1]))
+
+        # Step 3: Validation: column names must exactly match DAG variables
+        missing_columns = set(required_features) - set(X_df.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Missing required columns in input data: {list(missing_columns)}. "
+                f"DAG expects columns: {required_features}, but got: {list(X_df.columns)}"
+            )
+
+        return X_df[required_features]
+
+    def fit(self, X, y, sample_weight: Optional[Any] = None):
+        """
+        This method performs two-stage least squares regression using the specified causal graph.
+        It first fits the stage 1 estimator to predict the exposure variable from the instrument,
+        then fits the stage 2 estimator to predict the outcome variable from the predicted exposure 
+        and pretreatment variables.
+        """
+
+        # Step 0: validate Inputs
+        validate_data(
+            self,
+            X,
+            y,
+            accept_sparse=False,
+            ensure_2d=True,
+            ensure_min_features=2,
+            dtype="numeric",
+        )
+
+        # Step 1: Initialize data structures and read roles from DAG.
+
+        if self.stage1_estimator is None:
+            self.stage1_estimator = LinearRegression()
+        if self.stage2_estimator is None:
+            self.stage2_estimator = LinearRegression()
+   
+        stage1_estimator = clone(self.stage1_estimator)
+        stage2_estimator = clone(self.stage2_estimator)
+
+        # Step 1.1: Get roles from the causal graph and assign to attributes.
+        exposure_vars = self.causal_graph.get_role("exposure")
+        outcome_vars = self.causal_graph.get_role("outcome")
+        instrument_vars = self.causal_graph.get_role("instrument")
+
+        # Step 1.2: Validate that exactly one exposure, one outcome and atleast one instrument are specified.
+        if len(exposure_vars) != 1:
+            raise ValueError(
+                f"NaiveIVRegressor requires exactly one exposure; got {len(exposure_vars)}"
```

Comment:
The error message should be along the lines of: the current implementation only works for a single exposure and outcome.